cluster/engine.go

```diff
@@ -526,6 +525,10 @@ func (e *Engine) RemoveImage(name string, force bool) ([]types.ImageDelete, erro
 	rmOpts := types.ImageRemoveOptions{force, true}
 	dels, err := e.apiClient.ImageRemove(context.Background(), name, rmOpts)
 	e.CheckConnectionErr(err)
+	if err != nil {
+		return nil, err
```

Comment:
Looking at daemon code, yeah I don't think `ImageRemove` Is atomic, so I think you're right about refreshing the images :wink:.